scrapy/extensions/feedexport.py

```diff
@@ -470,8 +472,11 @@ def _settings_are_valid(self):
         If batch class or related contraints are specified, uri has to contain
         %(batch_time)s or %(batch_id)d to distinguish different files of partial output
         """
-        for uri_template, batch in self.batches.items():
-            if batch.enabled and not re.search(r'%\(batch_time\)s|%\(batch_id\)', uri_template):
+        for uri_template, values in self.feeds.items():
+            if (
+                ('batch' in values or self.batches[uri_template].enabled)
+                and not re.search(r'%\(batch_time\)s|%\(batch_id\)', uri_template)
+            ):
```

Comment:
Now that batch handlers do not need to define `enabled`, this probably needs an update.

You should also add a test that would have caught this issue.

Reply:
So what I am attempting here is giving only the default batch handler, an `enabled` attribute so as to know if any in built criteria has been activated. This is not needed in custom batch handlers. This I think is an implementation detail for the default batch handler only which also has `NotConfigured` exception for incorrect values.

Reply:
So I think it's now more closer to being a Scrapy component than before :sweat_smile:

Reply:
Actually now that I think about it we can remove the automatic default batch handler assignment for non batch feeds.

Reply:
`self.enable` will always be `True`, because in scenarios where it would be `False` a `NotConfigured` exception is raised, so the batch handler object is never created or used.